src/binary/index.js

```diff
@@ -1,15 +1,26 @@
 'use strict';
 
-const assertions = [
-  require('./equal'),
-  require('./numeric'),
-  require('./add'),
-  require('./relational'),
-];
+const assertions = {
+  equal: require('./equal'),
+  numeric : require('./numeric'),
+  add: require('./add'),
+  relational: require('./relational'),
+};
+
+function isExcluded(options, operator) {
+  return options.excludeOperators.indexOf(operator) >= 0;
+}
+
 
 module.exports = class BinaryAssertions {
   constructor(options) {
-    this._assertions = options.enabled ? assertions.map(Assertion => new Assertion(options)) : [];
+    this._assertions = [];
+    for (const operator of Object.keys(assertions)) {
+      const Assertion = assertions[operator];
+      if (options.enabled && !isExcluded(options, operator)) {
```

Comment:
I think here we can optimize and move `options.enabled` check out of loop. And move loop itself into separate method. Also we can put hanging `isExcluded` into class.
E.g.
```js
module.exports = class BinaryAssertions {
  constructor(options) {
    this._assertions = [];
    this._options = options;
    if (this._options.enabled) {
      this._fillAssertions();
    }
  }
  _fillAssertions() {
    for (const operator of Object.keys(assertions)) {
      if (!this._isExcludedOperator(operator)) {
        const Assertion = assertions[operator];
        this._assertions.push(new Assertion(this._options));
      }
    }
  }
  _isExcludedOperator(operator) {
    return this._options.excludeOperators.indexOf(operator) >= 0;
  }
```